tests/integration_test.go

```diff
@@ -872,21 +873,28 @@ func TestSuccess_ConcurrentSwaps(t *testing.T) {
 
 	for _, tc := range makerTests {
 		select {
-		case err = <-tc.errCh:
+		case err := <-tc.errCh:
 			assert.NoError(t, err)
 		default:
 		}
 	}
 
 	for _, tc := range takerTests {
 		select {
-		case err = <-tc.errCh:
+		case err := <-tc.errCh:
 			assert.NoError(t, err)
 		default:
 		}
 	}
 
+	idMap := make(map[string]bool) // IDs created this test
+	for _, mt := range makerTests {
+		idMap[mt.offerID] = true
+	}
 	offersAfter, err := bc.GetOffers()
 	require.NoError(t, err)
-	require.Equal(t, numConcurrentSwaps, len(offersBefore)-len(offersAfter))
+	for _, o := range offersAfter {
+		id := o.GetID().String()
+		require.Falsef(t, idMap[id], "offer id %s was not removed", id)
+	}
```

Comment:
This is the best idea I could come up with to deal with an offer showing up (presumably from a previously exited test), while this test is running. Other suggestions welcome.